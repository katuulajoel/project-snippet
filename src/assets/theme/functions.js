export const pxToRem = (pixels) => {
  const value = 1 * (pixels / 16);
  return `${value}rem`;
};

export const functions = {
  pxToRem,
};
