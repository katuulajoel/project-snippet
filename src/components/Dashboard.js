import React from "react";
import DashboardLayout from "../../layouts/DashboardLayout";

export default function Dashboard() {
  return (
    <DashboardLayout>
      <h3>Dashboard</h3>
    </DashboardLayout>
  );
}
