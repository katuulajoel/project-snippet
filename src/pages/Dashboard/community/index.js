import React from "react";
import DashboardLayout from "../../../layouts/DashboardLayout";

export default function Community() {
  return (
    <DashboardLayout>
      <h3>Community</h3>
    </DashboardLayout>
  );
}
