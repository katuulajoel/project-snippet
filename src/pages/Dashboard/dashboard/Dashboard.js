import React from "react";
import { Row, Col } from "reactstrap";
import moment from "moment";
import Avatar from "../../../components/Avatar";
import DashboardLayout from "../../../layouts/DashboardLayout";
import css from "./Dashboard.module.scss";
import SummaryPlaceholder from "../../../components/SummaryPlaceholder/SummaryPlaceholder";
import { isAdminOrPM, isDev, isPM } from "../../../components/utils/auth";

export default function Dashboard() {
  const user = {
    display_name: "Katuula Joel",
    avatar_url: "",
    welcomeMessage: "Welcome Back.",
  };
  return (
    <DashboardLayout>
      <Row>
        <Col sm={8}>
          <div className={`${css.card} ${css.header}`}>
            <Avatar image={user.avatar_url} size="dash" />
            <div className={css.profile}>
              <h3>Hi {user.display_name} üëãüèª</h3>

              <span className={css.subText}>{user.welcomeMessage}</span>
            </div>
            <div className={css.date}>
              <span className={css.subText}>
                {moment().format("ddd, Do MMM YYYY")}
              </span>
            </div>
          </div>
          <div className={css.card}>
            <h3 className={css.title}>
              Recent Notifications
              <small
                onClick={() => {}}
                style={{
                  float: "right",
                  cursor: "pointer",
                }}
              >
                Clear all
              </small>
            </h3>

            <SummaryPlaceholder description="No recent notifications" />
          </div>

          <div className={css.card}>
            <h3 className={css.title}>
              {`${isAdminOrPM() ? "" : "Your "}`}
              Active Projects
            </h3>

            <SummaryPlaceholder description="No active projects" />
          </div>
        </Col>
        <Col sm={4}>
          <div className={css.card}>
            {isDev() || isPM() ? (
              <>
                <h3 className={css.title}>Latest Reports</h3>

                <SummaryPlaceholder description="No upcoming updates" />
              </>
            ) : (
              <>
                <h3 className={css.title}>Latest reports</h3>

                <SummaryPlaceholder description="No upcoming updates" />
              </>
            )}
          </div>

          <div className={css.card}>
            <h3 className={css.title}>Upcoming Payments</h3>
            <SummaryPlaceholder description="No upcoming payments" />
          </div>
        </Col>
      </Row>
    </DashboardLayout>
  );
}
