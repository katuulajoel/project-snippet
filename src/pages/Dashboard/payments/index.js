import React, { useEffect } from "react";
import { useParams, useLocation } from "react-router-dom";
import styled from "styled-components";
import { Switch, Route, Redirect } from "react-router-dom";
import { useSelector, useDispatch } from "react-redux";

import Icon from "../../../components/Icon";
import SectionNav from "../../../components/SectionNav";
import { isDev } from "../../../components/utils/auth";
import DashboardLayout from "../../../layouts/DashboardLayout";
import InvoiceListContainer from "./InvoiceListContainer";
import Payments from "./Payments";

import Payouts from "./Payouts";
import { downloadCsv } from "./utils/paymentActions";
import PaymentTotals from "./components/PaymentTotals";
import { NavActions } from "./styles";

export default function PaymentsPage() {
  let { type } = useParams();
  let { pathname } = useLocation();
  const dispatch = useDispatch();
  const { csv } = useSelector(({ Invoice }) => Invoice);
  let filter = pathname.split("/")[3];

  useEffect(() => {
    if (csv) {
      var fileDownload = require("js-file-download");
      fileDownload(
        csv,
        `${filter?.toUpperCase()} ${type === "in" ? "Payments" : "Payouts"}.csv`
      );
      dispatch({ type: "CLEAR_CSV" });
    }
  }, [csv]);

  return (
    <DashboardLayout>
      <Wrapper>
        <PaymentTotals type={type} />
        <SectionNav
          links={[
            ...(isDev()
              ? [
                  { route: `payments/${type}/all`, name: "All" },
                  {
                    route: `payments/${type}/pending`,
                    name: "Pending",
                  },
                  { route: `payments/${type}/paid`, name: "Paid" },
                ]
              : [
                  { route: `payments/${type}/all`, name: "All" },
                  {
                    route: `payments/${type}/overdue`,
                    name: "Overdue",
                  },
                  {
                    route: `payments/${type}/pending`,
                    name: "Pending",
                  },
                  { route: `payments/${type}/paid`, name: "Paid" },
                  {
                    route: `payments/${type}/archived`,
                    name: "Archived",
                  },
                ]),
          ]}
          style={{ marginTop: "40px" }}
          rightActions={
            !isDev() && (
              <NavActions>
                <a
                  href="#"
                  className="add-btn"
                  onClick={() =>
                    downloadCsv(filter, type === "in" ? "Payments" : "Payouts")
                  }
                >
                  <Icon name="file-upload-outline" size="sm" /> Export
                </a>
              </NavActions>
            )
          }
        />
        <Switch>
          <Route
            path={`/payments/${type}/:filter`}
            render={(props) => {
              return (
                <InvoiceListContainer {...props} type={type}>
                  {type === "in" ? (
                    <Payments {...props} />
                  ) : (
                    <Payouts {...props} />
                  )}
                </InvoiceListContainer>
              );
            }}
          />
          <Redirect from="*" to={`/payments/${type}/all`} />
        </Switch>
      </Wrapper>
    </DashboardLayout>
  );
}

const Wrapper = styled.div`
  padding-top: 30px; /* TODO: moving this couldnt work cause navbar doesnt record right hieght on page reload */
`;
