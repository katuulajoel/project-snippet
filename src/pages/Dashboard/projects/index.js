/* eslint-disable no-unused-vars */
/* eslint-disable react/prop-types */
import React, { useEffect } from "react";
import { Redirect, Route, Switch } from "react-router";
import styled from "styled-components";
import { useSelector, useDispatch } from "react-redux";

import SummaryPlaceholder from "../../../components/SummaryPlaceholder/SummaryPlaceholder";
import { hasProjectAccess, isPM } from "../../../components/utils/auth";
import DashboardLayout from "../../../layouts/DashboardLayout";
import { fetchProject } from "../../../actions/ProjectActions";
import Progress from "../../../components/Progress";
import PaymentContainer from "./PaymentContainer";

const ProjectPage = ({ match }) => {
  const { isMakingRequest, project } = useSelector(({ Projects }) => Projects);
  const dispatch = useDispatch();

  useEffect(() => {
    console.log(project);
    if (project === null) {
      dispatch(fetchProject(21)); // TODO: project Id should gotten by selecting from list
    }
  }, []);

  if (isMakingRequest.fetch || project === null) {
    // TODO: if project is null show appropriate error
    return <Progress style={{ textAlign: "center" }} />;
  }

  return (
    <DashboardLayout>
      {hasProjectAccess(project) || isPM() ? (
        <Switch>
          {[
            ["docs", <></>],
            ["plan", <></>],
            [
              "pay",
              <PaymentContainer project={project.project} key={"payment"} />,
            ],
            ["reports", <></>],
            ["settings", <></>],
          ].map((path) => {
            return (
              <Route
                key={`project-management-path--${path}`}
                path={`${match.path}/${project.project.id}/${path[0]}`}
                render={() => path[1]}
              />
            );
          })}
          <Redirect from="*" to={`/projects/${project.project.id}/plan`} />
        </Switch>
      ) : (
        <SummaryPlaceholder
          className="page"
          description={`You don't have permission to access this project's resources`}
        />
      )}
    </DashboardLayout>
  );
};

export default ProjectPage;
