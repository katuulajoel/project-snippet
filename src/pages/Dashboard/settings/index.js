import React from "react";
// import { useDispatch, useSelector } from "react-redux";
import { Switch, Route } from "react-router-dom";
import DashboardLayout from "../../../layouts/DashboardLayout";
import CreateUser from "./Invite/CreateUser";
import Invite from "./Invite/InviteUser";
// import Invite from "./Invite";
import PendingInvite from "./Invite/PendingInvites";
// import * as userActions from "../../../actions/UserActions";

export default function Settings() {
  return (
    <DashboardLayout>
      <div className={`content-card settings-card settings-container clearfix`}>
        <Switch>
          <Route exact path="/settings/invite" component={Invite} />
          <Route exact path="/settings/invite/create" component={CreateUser} />
          <Route
            exact
            path="/settings/invite/pending"
            component={PendingInvite}
          />
        </Switch>
      </div>
    </DashboardLayout>
  );
}
