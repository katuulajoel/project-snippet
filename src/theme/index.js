import { functions } from "./functions";
import { mixins } from "./mixins";
import { variables } from "./variables";

export default {
  ...variables,
  mixins: mixins,
  functions: functions,
};
