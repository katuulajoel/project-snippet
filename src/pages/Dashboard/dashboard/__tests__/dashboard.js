import React from "react";
import renderer from "react-test-renderer";
import Dashboard from "../Dashboard";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { ThemeProvider } from "styled-components";
import theme from "../../../../theme";

const middlewares = [thunk];

const mockAppStore = (state) => {
  const mockStore = configureStore(middlewares);
  return mockStore(state);
};

describe("Dashboard test", () => {
  it("Should match snapshot test", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={mockAppStore()}>
            <ThemeProvider theme={theme}>
              <Dashboard />
            </ThemeProvider>
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
