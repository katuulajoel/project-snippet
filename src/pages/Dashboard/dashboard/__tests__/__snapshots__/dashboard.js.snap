// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard test Should match snapshot test 1`] = `
Array [
  <nav
    className="sc-pNWdM ixJkbw navbar "
  >
    <div
      className="title-bar"
    >
      <a
        className="navbar-brand"
        href="/dashboard"
        onClick={[Function]}
      >
        Dashboard
      </a>
      <ul
        className="navbar-nav ml-auto"
      >
        <li>
          <div
            className="sc-eCApnc WEWHt"
          >
            <div
              className="input-group sc-jSFjdj hTzrBi input-search"
            >
              <span
                className="input-group-prepend"
              >
                <span
                  className="input-group-text"
                >
                  <i
                    className="tg-ic-search1  "
                  />
                </span>
              </span>
              <div
                className="sc-gtsrHT jGBtBH"
                data-testid="input"
              >
                <input
                  autoComplete="off"
                  className="form-control"
                  data-testid=""
                  id=""
                  label="Label"
                  name="search"
                  onChange={[Function]}
                  placeholder="Search...."
                  type="input"
                  value=""
                />
              </div>
              <span
                className="input-group-prepend"
              />
            </div>
            
          </div>
        </li>
      </ul>
    </div>
    <div
      className="sc-fujyAs cKNmWV"
    />
  </nav>,
  <div
    className="sc-jrsJWt ilZZuz sidebar"
    id="sidebar"
  >
    <img
      alt="company-logo"
      src={Object {}}
    />
    <ul
      className="nav"
    >
      <li>
        <a
          aria-current={null}
          href="/dashboard"
          onClick={[Function]}
        >
          <span
            className="menu-icon"
          >
            <i
              className="tg-ic-outline-dashboard tunga-ic-sz-sidebar "
            />
          </span>
          <span>
            Dashboard
          </span>
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/projects"
          onClick={[Function]}
        >
          <span
            className="menu-icon"
          >
            <i
              className="tg-ic-baseline-folder-open tunga-ic-sz-sidebar "
            />
          </span>
          <span>
            Projects
          </span>
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/payments"
          onClick={[Function]}
        >
          <span
            className="menu-icon"
          >
            <i
              className="tg-ic-round-payment tunga-ic-sz-sidebar "
            />
          </span>
          <span>
            Payments
          </span>
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/settings"
          onClick={[Function]}
        >
          <span
            className="menu-icon"
          >
            <i
              className="tg-ic-outline-settings tunga-ic-sz-sidebar "
            />
          </span>
          <span>
            Settings
          </span>
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/community"
          onClick={[Function]}
        >
          <span
            className="menu-icon"
          >
            <i
              className="tg-ic-baseline-folder-open tunga-ic-sz-sidebar "
            />
          </span>
          <span>
            Community
          </span>
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          data-testid="signout"
          href="/logout"
          onClick={[Function]}
        >
          <span
            className="menu-icon"
          >
            <i
              className="tg-ic-log-out tunga-ic-sz-sidebar "
            />
          </span>
          <span>
            Log Out
          </span>
        </a>
      </li>
    </ul>
  </div>,
  <div
    className="sc-kEqXSa gCTtLP"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-8"
      >
        <div
          className="undefined undefined"
        >
          <div
            className="sc-bdnxRM eNsiXn avatar-initials"
          >
            <div
              className="avatar avatar-initials"
              style={Object {}}
            >
              KJ
            </div>
          </div>
          <div>
            <h3>
              Hi 
              Katuula Joel
               üëãüèª
            </h3>
            <span>
              Welcome Back.
            </span>
          </div>
          <div>
            <span>
              Thu, 2nd Sep 2021
            </span>
          </div>
        </div>
        <div
          className="dashboard-card"
        >
          <h3
            className="title"
          >
            Recent Notifications
            <small
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "float": "right",
                }
              }
            >
              Clear all
            </small>
          </h3>
          <div
            className="undefined undefined"
          >
            <img
              alt="empty file image"
              src={Object {}}
            />
            <p>
              No recent notifications
            </p>
          </div>
        </div>
        <div
          className="dashboard-card"
        >
          <h3
            className="title"
          >
            Your 
            Active Projects
          </h3>
          <div
            className="content"
          />
        </div>
      </div>
      <div
        className="col-sm-4"
      >
        <div
          className="dashboard-card"
        >
          <h3
            className="title"
          >
            Latest reports
          </h3>
          <div
            className="content"
          />
        </div>
        <div
          className="dashboard-card"
        >
          <h3
            className="title"
          >
            Latest reports
          </h3>
          <div
            className="content"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;
